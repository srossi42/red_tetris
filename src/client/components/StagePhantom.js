import React from 'react';

import CellPhantom from './CellPhantom';

const StagePhantom = ({ stage }) => {

    const TETROMINOS = {
        0: { color: "128, 128, 128" },
        I: { color: "80, 227, 230" },
        J: { color: "36, 95, 223" },
        L: { color: "223, 173, 36" },
        O: { color: "223, 217, 36" },
        S: { color: "48, 211, 56" },
        T: { color: "132, 61, 198" },
        Z: { color: "227, 78, 78" }
      };

    return stage.map(row => row.map((cell, x) =>
        <CellPhantom key={x} type={cell[0]} color={TETROMINOS[cell[0]].color}/>
    ))
}



export default StagePhantom