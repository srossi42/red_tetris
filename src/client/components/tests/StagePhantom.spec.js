import React from 'react'
import { shallow } from 'enzyme';
import StagePhantom from '../StagePhantom'
import { createStage } from '../../gameHelper'
import toJson from "enzyme-to-json"

describe('<StagePhantom/> Component', () => {
  it('rendering correctly StagePhantom', () => {
    let stage = createStage()
    let wrapper = shallow(
      <StagePhantom stage={stage}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
