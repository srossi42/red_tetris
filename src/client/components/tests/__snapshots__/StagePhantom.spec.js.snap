// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StagePhantom/> Component rendering correctly StagePhantom 1`] = `
Array [
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="0"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="1"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="2"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="3"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="4"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="5"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="6"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="7"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="8"
    type={0}
  />,
  <Memo(CellPhantom)
    color="128, 128, 128"
    key="9"
    type={0}
  />,
]
`;
